libraries/ESP8266mDNS/src/LEAmDNS2Host_Transfer.cpp

```diff
@@ -0,0 +1,2470 @@
+            DEBUG_OUTPUT.printf_P(PSTR(" Type:%s Class:%s\n"), _RRType2Name(p_rRRQuestion.m_Header.m_Attributes.m_u16Type), _RRClass2String(p_rRRQuestion.m_Header.m_Attributes.m_u16Class, true));
+        );
+    }
+    DEBUG_EX_ERR(if (!bResult) DEBUG_OUTPUT.printf_P(PSTR("%s _readRRQuestion: FAILED!\n"), _DH()););
+    return bResult;
+}
+
+/*
+    MDNSResponder::_readRRAnswer
+
+    Reads an answer (eg. _http._tcp.local PTR OP TTL MyESP._http._tcp.local)
+    from the UDP input buffer.
+    After reading the domain and type info, the further processing of the answer
+    is transferred the answer specific reading functions.
+    Unknown answer types are processed by the generic answer reader (to remove them
+    from the input buffer).
+
+*/
+bool clsLEAMDNSHost::_readRRAnswer(clsLEAMDNSHost::clsRRAnswer*& p_rpRRAnswer)
+{
+    //DEBUG_EX_INFO(DEBUG_OUTPUT.printf_P(PSTR("%s _readRRAnswer\n")););
+
+    bool    bResult = false;
+
+    clsRRHeader    header;
+    uint32_t            u32TTL;
+    uint16_t            u16RDLength;
+    if ((_readRRHeader(header)) &&
+            (_udpRead32(u32TTL)) &&
+            (_udpRead16(u16RDLength)))
+    {
+        /*  DEBUG_EX_INFO(
+                DEBUG_OUTPUT.printf_P(PSTR("%s _readRRAnswer: Reading 0x%04X answer (class:0x%04X, TTL:%u, RDLength:%u) for "), header.m_Attributes.m_u16Type, header.m_Attributes.m_u16Class, u32TTL, u16RDLength);
+                _printRRDomain(header.m_Domain);
+                DEBUG_OUTPUT.printf_P(PSTR("\n"));
+                );*/
+
+        switch (header.m_Attributes.m_u16Type /*& (~0x8000)*/)      // Topmost bit might carry 'cache flush' flag
+        {
+#ifdef MDNS_IPV4_SUPPORT
+        case DNS_RRTYPE_A:
+            p_rpRRAnswer = new clsRRAnswerA(header, u32TTL);
+            bResult = _readRRAnswerA(*(clsRRAnswerA*&)p_rpRRAnswer, u16RDLength);
+            break;
+#endif
+        case DNS_RRTYPE_PTR:
+            p_rpRRAnswer = new clsRRAnswerPTR(header, u32TTL);
+            bResult = _readRRAnswerPTR(*(clsRRAnswerPTR*&)p_rpRRAnswer, u16RDLength);
+            break;
+        case DNS_RRTYPE_TXT:
+            p_rpRRAnswer = new clsRRAnswerTXT(header, u32TTL);
+            bResult = _readRRAnswerTXT(*(clsRRAnswerTXT*&)p_rpRRAnswer, u16RDLength);
+            break;
+#ifdef MDNS_IPV6_SUPPORT
+        case DNS_RRTYPE_AAAA:
+            p_rpRRAnswer = new clsRRAnswerAAAA(header, u32TTL);
+            bResult = _readRRAnswerAAAA(*(clsRRAnswerAAAA*&)p_rpRRAnswer, u16RDLength);
+            break;
+#endif
+        case DNS_RRTYPE_SRV:
+            p_rpRRAnswer = new clsRRAnswerSRV(header, u32TTL);
+            bResult = _readRRAnswerSRV(*(clsRRAnswerSRV*&)p_rpRRAnswer, u16RDLength);
+            break;
+        default:
+            p_rpRRAnswer = new clsRRAnswerGeneric(header, u32TTL);
+            bResult = _readRRAnswerGeneric(*(clsRRAnswerGeneric*&)p_rpRRAnswer, u16RDLength);
+            break;
+        }
```

Comment:
It would help readability if this switch were put in its own function, e.g.: readAnswerType(). Two returns would be needed: bResult and p_rpRRAnswer.